File: conformance/bundles/05-ambiguous-match/cukes.steps.rs
//! Rust sibling of `cukes.steps.ts` (bundle `05-ambiguous-match`).

use varar::{Steps, Value};

pub fn register(s: &mut Steps) {
    // Both expressions match "I have 5 cukes" → ambiguous-match diagnostic.
    s.stimulus("I have {int} cukes", |_state, _n| Ok(None));
    s.stimulus("I have 5 cukes", |_state| Ok(None));
}

pub fn state() -> Value {
    Value::Null
}

File: conformance/bundles/05-ambiguous-match/cukes.steps.cs
// C# sibling of cukes.steps.ts / .rs (bundle 05-ambiguous-match).
using Varar;
using Varar.Core;

namespace Varar.Corpus.B05;

public static class CukesSteps
{
    public static void Register(Steps s)
    {
        // Both expressions match "I have 5 cukes" → ambiguous-match diagnostic.
        s.Stimulus("I have {int} cukes", (state, n) => null);
        s.Stimulus("I have 5 cukes", state => null);
    }

    public static Value State() => Value.Null;
}
